kaspad/src/main.rs

```diff
@@ -151,31 +152,34 @@ pub fn main() {
         NetworkType::Simnet => unimplemented!("simnet params"),
     };
 
-    let (consensus_send, consensus_recv) = unbounded::<ConsensusEvent>();
-    let (event_processor_send, event_processor_recv) = unbounded::<Notification>();
+    let (notification_send, notification_recv) = unbounded();
 
     // Use `num_cpus` background threads for the consensus database as recommended by rocksdb
     let consensus_db = database::prelude::open_db(consensus_db_dir, true, num_cpus::get());
-    let consensus = Arc::new(Consensus::new(consensus_db, &config, consensus_send));
+    let consensus = Arc::new(Consensus::new(consensus_db, &config, notification_send));
     let monitor = Arc::new(ConsensusMonitor::new(consensus.processing_counters().clone()));
 
-    let utxoindex: DynUtxoIndexApi = if args.utxoindex {
+    let notify_service = Arc::new(NotifyService::new(consensus.notification_root(), notification_recv));
+
+    let index_service: Option<Arc<IndexService>> = if args.utxoindex {
         // Use only a single thread for none-consensus databases
         let utxoindex_db = database::prelude::open_db(utxoindex_db_dir, true, 1);
-        Some(UtxoIndex::new(consensus.clone(), utxoindex_db).unwrap())
+        let utxoindex: DynUtxoIndexApi = Some(UtxoIndex::new(consensus.clone(), utxoindex_db).unwrap());
+        Some(Arc::new(IndexService::new(notify_service.clone(), utxoindex)))
     } else {
         None
     };
 
-    let event_processor = Arc::new(EventProcessor::new(utxoindex.clone(), consensus_recv, event_processor_send));
-
-    let rpc_core_server = Arc::new(RpcCoreServer::new(consensus.clone(), utxoindex, event_processor_recv));
+    let rpc_core_server = Arc::new(RpcCoreServer::new(consensus.clone(), notify_service.clone(), index_service.clone()));
```

Comment:
Same here. Pass the inner notifiers and `DynUtxoindex` if needed, and not the services themselves